packages/types/src/factors.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  EmailCodeStrategy,
+  EmailLinkStrategy,
+  OAuthStrategy,
+  PasswordStrategy,
+  PhoneCodeStrategy,
+  Web3Strategy,
+} from './strategies';
+
+export type EmailCodeFactor = {
+  strategy: EmailCodeStrategy;
+  email_address_id: string;
+  safe_identifier: string;
+};
+
+export type EmailLinkFactor = {
+  strategy: EmailLinkStrategy;
+  email_address_id: string;
+  safe_identifier: string;
+};
+
+export type PhoneCodeFactor = {
+  strategy: PhoneCodeStrategy;
+  phone_number_id: string;
+  safe_identifier: string;
+  default?: boolean;
+};
+
+export type Web3SignatureFactor = {
+  strategy: Web3Strategy;
+  web3_wallet_id: string;
+};
+
+export type PasswordFactor = {
+  strategy: PasswordStrategy;
+};
+
+export type OauthFactor = {
+  strategy: OAuthStrategy;
+};
+
+export type EmailCodeConfig = Omit<EmailCodeFactor, 'safe_identifier'>;
```

Comment:
❓ Is this the `EmailCodeConfig` or is it `EmailCodeParams`?

Reply:
Right now the mapping is:
- signIn.firstFactors  -> `EmailCodeFactor | EmailLinkFactor | ...`, used to describe the factor
- signIn.prepareFirstFactor -> `EmailLinkConfig | EmailCodeConfig | ...`, used to select a factor
- signIn.attemptFirstFactor -> `EmailCodeAttempt | EmailLinkAttempt | ...`, used to attempt a factor

But I'm not 100% sure about the `*Config` naming for the 2nd case. Would `EmailCodeFactorSelector` make more sense? Any naming suggestions are welcome :)